src/RequestHandle.cpp

```diff
@@ -175,6 +220,14 @@ auto RequestHandle::SetVersion(
     }
 }
 
+auto RequestHandle::SetMaxDownloadBytes(ssize_t max_download_bytes) -> void
+{
+    m_max_download_bytes = max_download_bytes;
+    m_bytes_written = 0;
+
+    init();
```

Comment:
hmm, this is the opposite of what I suggested, this will do a lot of duplicate work by now resetting a lot of fields on the curl handle that are unrelated to the progress/xfer functions.  I would like to see the reverse as if this is getting set on thousands of handles it _is_ a performance concern

Reply:
Okay, I see where I got confused. This should be correct now.